src/llvm-alloc-opt.cpp

```diff
@@ -1067,8 +1076,7 @@ void Optimizer::splitOnStack(CallInst *orig_inst)
                     store_ty = T_pjlvalue;
                 }
                 else {
-                    store_ty = PointerType::getWithSamePointeeType(
-                        T_pjlvalue, store_ty->getPointerAddressSpace());
+                    store_ty = PointerType::get(T_pjlvalue->getContext(), store_ty->getPointerAddressSpace());
```

Comment:
@maleadt We refactored this line to merge the code from #54113 but now the `gc-invariant-verifier` test is failing https://buildkite.com/julialang/julia-master/builds/35738#018f0295-5279-4b99-818d-36905524f086

Would you mind checking it out please?

Reply:
done!